src/SpectatorView.Unity/Assets/SpectatorView/Scripts/Compositor/CompositionManager.cs

```diff
@@ -638,6 +640,13 @@ private void StopRecordingAudio()
                 audioMemoryStream = null;
             }
         }
+        
+        private string ReduceFilePath(string filePath)
+        {
+            const string mp4Ext = ".mp4";
+            filePath = filePath.Trim().Replace("\\\\", "\\");
+            return filePath.Substring(0, filePath.LastIndexOf(mp4Ext) + mp4Ext.Length);

```

Comment:
I'm confused about why this is necessary:
1. When are there double "\\" instances in the path? Is that indicative of a bug in the native code that builds the path?
2. Why is there a need to trim a substring that cuts something after the ".mp4" in the path?

Reply:
this is a problem with the memory that is handed around, in the way it exists today we don't end up with a terminated string.

Reply:
i think your string builder suggestion will fix this problem.